src/dal/weblinklistener.ts

```diff
@@ -0,0 +1,134 @@
+import {inject, injectable} from "inversify";
+import {Logger} from "../bll/utils/logger";
+import {Disposable} from "vscode";
+import {HttpHostRequest} from "../bll/weblinklistener/httphostrequest";
+import {TYPES} from "../bll/utils/constants";
+import {Server, Socket, createServer} from "net";
+import {Utils} from "../bll/utils/utils";
+
+@injectable()
+export class WebLinkListener implements Disposable {
+
+    private readonly server: Server;
+    private readonly clients: Array<Socket> = [];
+    private static readonly ENCODING = "ascii";
+    private static readonly SOCKET_NUMBER_START: number = 63330;
+    private static readonly NUMBER_OF_LISTENERS: number = 10;
+    private static readonly SOCKET_NUMBER_END: number = WebLinkListener.SOCKET_NUMBER_START + 9;
+    private static readonly LOOPBACK: string = "127.0.0.1";
+
+    private static readonly CONTENT_TYPE: Buffer = new Buffer("Content-Type: image/gif\r\n", WebLinkListener.ENCODING);
+    private static readonly CONTENT_LENGTH: Buffer = new Buffer("Content-Length: ", WebLinkListener.ENCODING);
+    private static readonly HEADER_END: Buffer = new Buffer("\r\n\r\n", WebLinkListener.ENCODING);
+    private static readonly CONNECTION_CLOSE: Buffer = new Buffer("Connection: close\r\n", WebLinkListener.ENCODING);
+    private static readonly OK: Buffer = new Buffer(" 200 OK\r\n", WebLinkListener.ENCODING);
+    private static readonly SERVER: Buffer = new Buffer("Server: Visual Studio Code Activation 1.0\r\n",
+                                                        WebLinkListener.ENCODING);
+
+    public static readonly SUCCESS_STREAM: Uint8Array = new Uint8Array([
+            0x47, 0x49, 0x46, 0x38, 0x39, 0x61, 0x02, 0x00,
+            0x02, 0x00, 0x80, 0xFF, 0x00, 0xFF, 0xFF, 0xFF,
+            0x00, 0x00, 0x00, 0x2C, 0x00, 0x00, 0x00, 0x00,
+            0x01, 0x00, 0x01, 0x00, 0x00, 0x02, 0x02, 0x44,
+            0x01, 0x00, 0x3B
+        ]);
+
+    public static readonly FAILURE_STREAM: Uint8Array = new Uint8Array([
+            0x47, 0x49, 0x46, 0x38, 0x39, 0x61, 0x01, 0x00,
+            0x01, 0x00, 0x80, 0xFF, 0x00, 0xFF, 0xFF, 0xFF,
+            0x00, 0x00, 0x00, 0x2C, 0x00, 0x00, 0x00, 0x00,
+            0x01, 0x00, 0x01, 0x00, 0x00, 0x02, 0x02, 0x44,
+            0x01, 0x00, 0x3B
+        ]);
+
+    public constructor(@inject(TYPES.HttpHostRequest) private httpHostRequest: HttpHostRequest) {
+        this.server = createServer();
```

Comment:
It looks like it would be simpler to use [http.Server](https://nodejs.org/api/http.html#http_http_createserver_options_requestlistener) instead of manually implementing it, example: https://nodejs.org/en/docs/guides/anatomy-of-an-http-transaction/